nixos/modules/services/misc/mame.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mame;
+  mame = "mame${lib.optionalString pkgs.stdenv.is64bit "64"}";
+in
+{
+  options = {
+    services.mame = {
+      enable = mkEnableOption "MAME networking";
+      user = mkOption {
+        type = types.str;
+        description = "User under which MAME is run";
+      };
+      emuAddr = mkOption {
+        type = types.str;
+        description = "IP address you use inside the virtual machine";
+      };
+      hostAddr = mkOption {
+        type = types.str;
+        description = "IP address of host network interface";
+      };
+    };
+  };
```

Comment:
I'm confused on what this service exactly does. What is this mentioned virtual machine? why does the script chmod 666 the tun device? how does networkling happen afterwards?

Reply:
@Lassulus Its a service for mame package (mame is an emulator of old machines). It creates a virtual network device which mame then uses. Without a service network under mame will not work.

Reply:
Well, I get what the goal is, I'm skeptical of the implementation, for example, I wouldn't know where to get the hostAddr or the emuAddr from if I would want to use this.

Reply:
@Lassulus I've added more detailed descriptions. Implementation of the `taputil.sh` is up to mame team. Module just adds required capabilities to mame binary and moves knobs from the script level to nixos config level.